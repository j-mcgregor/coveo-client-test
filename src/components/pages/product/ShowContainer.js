import React from 'react';
import { Link } from 'react-router-dom';

import PropTypes from 'prop-types';

const ShowContainer = ({ product }) => {
  const { title, excerpt } = product;

  return (
    <div className="container grid-md">
      <div className="columns">
        <div className="column">
          <Link to="/">Back</Link>
        </div>
      </div>
      <div className="columns">
        <div className="column">
          <h1>{title}</h1>
        </div>
      </div>
      <div className="columns">
        <div className="column col-4 col-md-6 col-sm-12">
          <small>{excerpt}</small>
        </div>
        <div className="divider-vert" data-content="" />
      </div>
    </div>
  );
};

ShowContainer.propTypes = {
  product: PropTypes.shape({}),
  match: PropTypes.shape({
    params: PropTypes.shape({
      title: PropTypes.string
    })
  })
};

ShowContainer.defaultProps = {
  product: {},
  match: {}
};

export default ShowContainer;
