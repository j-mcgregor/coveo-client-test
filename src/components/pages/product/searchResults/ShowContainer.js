import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import ProductBanner from './sections/ProductBanner';
import SuggestedList from './sections/SuggestedList';

export const mapUsersAndComments = (users = [], comments = []) =>
  users.map(user => {
    const userComments = comments.filter(comment => {
      if (comment.postId === user.id) {
        return {
          id: comment.id,
          body: comment.body
        };
      }
    });
    return {
      id: user.id,
      name: user.name,
      email: user.email,
      comments: userComments
    };
  });

const ShowContainer = ({
  results,
  selected,
  fetchUsers,
  fetchComments,
  selectProduct
}) => {
  useEffect(() => {
    fetchComments();
    fetchUsers();
  }, []);

  return (
    <div>
      <ProductBanner selected={selected} />
      <SuggestedList results={results} selectProduct={selectProduct} />
    </div>
  );
};

ShowContainer.propTypes = {
  results: PropTypes.arrayOf(PropTypes.shape({})),
  selected: PropTypes.shape({}),
  match: PropTypes.shape({
    params: PropTypes.shape({
      title: PropTypes.string
    })
  }),
  users: PropTypes.shape({}),
  comments: PropTypes.shape({}),
  fetchUsers: PropTypes.func,
  fetchComments: PropTypes.func,
  selectProduct: PropTypes.func.isRequired
};

ShowContainer.defaultProps = {
  results: [],
  selected: {},
  match: {},
  users: {},
  comments: {},
  fetchUsers: () => {},
  fetchComments: () => {}
};

export default ShowContainer;
