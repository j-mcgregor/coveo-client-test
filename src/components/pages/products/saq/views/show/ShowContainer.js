import PropTypes from 'prop-types';
import React from 'react';
import ProductBanner from '../../sections/ProductBanner';
import SuggestedList from '../../sections/SuggestedList';
import { ThemeType } from '../../../../../types';

const ShowContainer = ({ results, selected, selectProduct, theme }) => {
  const palette = theme.theme;
  return (
    <div>
      <ProductBanner selected={selected} />
      <SuggestedList
        results={results}
        selectProduct={selectProduct}
        title="Suggested results"
        palette={palette}
      />
    </div>
  );
};

ShowContainer.propTypes = {
  results: PropTypes.arrayOf(PropTypes.shape({})),
  selected: PropTypes.shape({}),
  selectProduct: PropTypes.func.isRequired,
  theme: ThemeType
};

ShowContainer.defaultProps = {
  results: [],
  selected: {},
  theme: {}
};

export default ShowContainer;
