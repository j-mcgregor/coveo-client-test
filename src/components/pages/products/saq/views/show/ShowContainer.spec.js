/* eslint-disable react/prop-types */
import React from 'react';
import { shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import ShowContainer from './ShowContainer';

describe('ShowContainer', () => {
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      selectProduct: jest.fn(),
      results: [],
      selected: {}
    };
  });

  it('should render', () => {
    const wrapper = shallow(
      <ThemeProvider theme={{ light: { color: 'yellow' } }}>
        <ShowContainer {...defaultProps} />
      </ThemeProvider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
