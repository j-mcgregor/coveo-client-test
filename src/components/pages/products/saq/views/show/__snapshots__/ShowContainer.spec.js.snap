// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowContainer should render 1`] = `ShallowWrapper {}`;
